src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs

```diff
@@ -96,40 +101,46 @@ void INotifyInitialized.Initialized()
         => this.Start();
 
     [ComputeMethod]
-    public virtual async Task<IReadOnlyList<ChatInfo>> List(CancellationToken cancellationToken = default)
+    public virtual async Task<int> GetChatCount(CancellationToken cancellationToken = default)
     {
         var settings = await ListSettings.Use(cancellationToken).ConfigureAwait(false);
-        DebugLog?.LogDebug("-> List: {Settings}", settings);
-        var filterId = settings.Filter.Id;
-        var filteredChats = await ListFiltered(filterId, cancellationToken).ConfigureAwait(false);
-        var preOrderedChats = filteredChats
-            .OrderByDescending(c => c.Contact.IsPinned)
-            .ThenByDescending(c => c.HasUnreadMentions);
-        var orderedChats = settings.Order switch {
-            ChatListOrder.ByLastEventTime => preOrderedChats
-                .ThenByDescending(c => c.News.LastTextEntry?.Version ?? 0),
-            ChatListOrder.ByOwnUpdateTime => preOrderedChats
-                .ThenByDescending(c => c.Contact.TouchedAt),
-            ChatListOrder.ByUnreadCount => preOrderedChats
-                .ThenByDescending(c => c.UnreadCount.Value)
-                .ThenByDescending(c => c.News.LastTextEntry?.Version),
-            ChatListOrder.ByAlphabet => filteredChats
-                .OrderByDescending(c => c.Contact.IsPinned)
-                .ThenBy(c => c.Chat.Title, StringComparer.Ordinal),
-            _ => throw new ArgumentOutOfRangeException(nameof(settings)),
-        };
-        var result = orderedChats.ToList();
-        DebugLog?.LogDebug("<- List: {Settings}", settings);
-        return result;
+        var filteredChats = await ListFiltered(settings.Filter.Id, cancellationToken).ConfigureAwait(false);
+        return filteredChats.Count;
     }
 
     [ComputeMethod]
-    protected virtual async Task<IReadOnlyList<ChatInfo>> ListFiltered(Symbol filterId, CancellationToken cancellationToken)
+    public virtual Task<ChatId?> GetChatId(int position, CancellationToken cancellationToken = default)
+    {
+        ChatId? chatId;
+        lock (_chatPositionLock) {
+            if (position >= _chatPositionMap.Count)
+                return Task.FromResult<ChatId?>(null);
+
+            chatId = _chatPositionMap[position];
```

Comment:
`_chatPositionMap` is an amazing name for the list.

Maybe `_chatIdByPosition`?

Reply:
IMO event `_chatIds` would be a better choice...

Reply:
Don't think that each map is a dictionary